test/functionalTest/cases/3004_v2_reg_create/v2_reg_create_errors.test

```diff
@@ -0,0 +1,1648 @@
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+Errors during registration creation in APIv2
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB 0-255 IPv4 -statCounters
+
+--SHELL--
+
+# 01. Invalid JSON in payload
+# 02. Payload is not a JSON object
+# 03. Payload is an empty object
+# 04. description is not a string
+# 05. description too long
+# 06. forbidden characters in description
+# 07. dataProvided missing
+# 08. dataProvided not a JSON object
+# 09. dataProvided empty
+# 10. dataProvided:entities not an array
+# 11. dataProvided:entities:entity not an object
+# 12. dataProvided:entities:entity without id and idPattern
+# 13. dataProvided:entities:entity with both id and idPattern
+# 14. dataProvided:entities:entity:id not a string
+# 15. dataProvided:entities:entity:id given but empty
+# 16. dataProvided:entities:entity:id with forbidden characters
+# 17. dataProvided:entities:entity:id too long
+# 18. dataProvided:entities:entity:idPattern not a string
+# 19. dataProvided:entities:entity:idPattern given but empty
+# 20. dataProvided:entities:entity with both type and typePattern
+# 21. dataProvided:entities:entity:type not a string
+# 22. dataProvided:entities:entity:type given but empty
+# 23. dataProvided:entities:entity:type with forbidden characters
+# 24. dataProvided:entities:entity:type too long
+# 25. dataProvided:entities:entity:typePattern not a string
+# 26. dataProvided:entities:entity:typePattern given but empty
+# 27. dataProvided:entities:attrs not an array
+# 28. dataProvided:entities:attrs item not a string
+# 29. dataProvided:entities:attrs item empty string
+# 30. provider missing
+# 31. provider empty
+# 32. provider not a JSON object
+# 33. provider has no http field
+# 34. supportedForwardingMode not a string
+# 35. invalid supportedForwardingMode
+# 36. provider/http not a JSON object
+# 37. provider/http/url missing
+# 38. provider/http/url not a string
+# 39. provider/http/url is the empty string
+# 40. provider/http/url invalid
+# 41. status is not a string
+# 42. invalid value for status
+# 43. legacyForwarding missing
+# 44. legacyForwarding not a boolean
+# 45. legacyForwarding set to false
```

Comment:
Maybe 45-48 should be moved to a different .test. They are somehow temporal, so when they get implemented it is a matter of remove that .test and let the present one untouched.

Reply:
Or, we just wait until they fail and remove them then. Easier

NTC